### PR TITLE
Merge seek 1.15.1 into LDH

### PR DESCRIPTION
**Pull request by ReneHaensel**

### CODE REVIEW

**Comment by whomingbird** [comment]
could you please describe the new features of the PR?

File: app/controllers/projects_controller.rb [hunk 1]
```diff
@@ -543,77 +543,206 @@ def order_investigations
...
-    user_json=@user_current.to_json
-    project_attributes_json_parse=JSON.parse(project_attributes_json)
-
-    meta_data=JSON.parse(@project.to_json(only:[:created_at,:updated_at, :uuid]))
-    meta_data=meta_data.transform_keys { |key| key == "created_at" ? "created" : key }
-    meta_data=meta_data.transform_keys { |key| key == "updated_at" ? "modified" : key }
+    project_attributes_json = ProjectSerializer.new(@project).to_json
+
+    @user_current = current_person
+    special_characters = {
```
Comment (whomingbird) [inline app/controllers/projects_controller.rb:549]: can this move to a function ?

File: app/controllers/projects_controller.rb [hunk 2]
```diff
@@ -543,77 +543,206 @@ def order_investigations
 
   def publish_to_csh
```
Comment (whomingbird) [inline app/controllers/projects_controller.rb:544]: i think the most of code of the "publish_to_csh" should be move to a module in `lib/nfdi4health`,  the code for preparing and parsing json could be reused by other resource to publish to CSH.

File: app/views/projects/_buttons.html.erb [hunk 3]
```diff
@@ -63,21 +63,16 @@
 
       <%= order_icon(item,current_user, order_investigations_project_path(item), item.investigations, 'investigation') %>
     <% end %>
-    <% if Seek::Config.n4h_enabled %>
+    <% if Seek::Config.n4h_enabled && item.extended_metadata != nil %>
     <li><%= image_tag_for_key('publish', publish_to_csh_project_path(item), "Publish to NFDI4Health GCHSH", nil, "Publish to NFDI4Health GCHSH") -%></li>
     <% else %>
-      <% explanation = "The 'Publish to NFDI4Health GCHSH' item for #{text_for_resource(item).pluralize.downcase} in Server admin panel was deactivated by the admin." %>
+      <% explanation = "Your project does not include metadata, or The 'Publish to NFDI4Health GCHSH' item for #{text_for_resource(item).pluralize.downcase} in Server admin panel was deactivated by the admin." %>
```
Comment (whomingbird) [inline app/views/projects/_buttons.html.erb:69]: this should be moved to the `ProjectsHelper` 


`Seek::Config.n4h_enabled && item.extended_metadata != nil`, this code looks weird

**Review by whomingbird — COMMENTED** [review]
Main Concerns:

1. Modularization:
Move the publish_to_csh functionality to a module within lib/nfdi4health. This will enhance modularity and allow the code for preparing and parsing JSON to be reused by other resources needing to publish to CSH.
2. Testing:
Add tests to ensure the reliability and correctness of the publish_to_csh functionality. In the SEEK main branch, writing tests is a crucial part of SEEK development. Any feature cannot be merged into the main branch without proper tests.

File: app/controllers/projects_controller.rb (hunk 2 above)
Comment (MasoudAbedi) [reply app/controllers/projects_controller.rb:544]: https://github.com/nfdi4health/ldh/blob/52132af3321b9e76bed66eeeab1f07c428da8ad2/app/controllers/projects_controller.rb#L552

https://github.com/nfdi4health/ldh/blob/52132af3321b9e76bed66eeeab1f07c428da8ad2/app/controllers/investigations_controller.rb#L156

https://github.com/nfdi4health/ldh/blob/52132af3321b9e76bed66eeeab1f07c428da8ad2/app/controllers/studies_controller.rb#L364

**Pull request merged**